sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+        TOR A. This means if the Y cable is actively routing, the "check_active_linked_tor_side"
+        API will now return Tor A. It also implies that if the link is actively routing on this port, Y cable
+        MUX will start forwarding packets from TOR A to NIC, and drop packets from TOR B to NIC
+        regardless of previous forwarding state.
+
+        Args:
+            port:
+            an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            a Boolean, true if the toggle succeeded and false if it did not succeed.
+        """
+
+        raise NotImplementedError
+
+    def toggle_mux_to_torB(self, port):
+        """
+        This API specifically does a hard switch toggle of the Y cable's MUX regardless of link state to
+        TOR B. This means if the Y cable is actively routing, the "check_active_linked_tor_side"
+        API will now return Tor B. It also implies that if the link is actively routing on this port, Y cable
+        MUX will start forwarding packets from TOR B to NIC, and drop packets from TOR A to NIC
+        regardless of previous forwarding state.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            a Boolean, true if the toggle succeeded and false if it did not succeed.
+        """
+        raise NotImplementedError
+
+    def check_read_side(self, port):
+        """
+        This API specifically checks which side of the Y cable the reads are actually getting performed
+        from, either TOR A or TOR B or NIC and returns the value.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to Y end of a Y cable which can which side reading the MUX from
+
+        Returns:
+            an Integer, 1 if reading the Y cable from TOR A side(TOR 1).
+                      , 2 if reading the Y cable from TOR B side(TOR 2).
+                      , 0 if reading the Y cable from NIC side.
+                      , -1 if reading the Y cable API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_mux_direction(self, port):
+        """
+        This API specifically checks which side of the Y cable mux is currently point to
+        and returns either TOR A or TOR B. Note that this API should return mux-direction
+        regardless of whether the link is active/routing state or not.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            an Integer, 1 if the mux is pointing to TOR A .
+                      , 2 if the mux is pointing to TOR B.
+                      , -1 if checking which side mux is pointing to API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_active_linked_tor_side(self, port):
```

Comment:
How is this different from `check_mux_direction`?

Reply:
actually this takes into the account that link is active as well, if link is not active this will reflect as unknown, mux-direction on the other hand does not care about link-status